index.php

```diff
@@ -131,8 +134,32 @@
 /*
  * 3. Load some other essential includes.
  */
+$loader = require SMF\Config::$vendordir . '/autoload.php';
+$third_party_mappers = [];
+
+// Ensure $sourcedir is set to something valid.
+if (class_exists(Config::class, false) && isset(Config::$sourcedir)) {
+	$sourcedir = Config::$sourcedir;
+}
 
-require_once SMF\Config::$sourcedir . '/Autoloader.php';
+if (empty($sourcedir) || !is_dir($sourcedir)) {
+	$sourcedir = __DIR__;
+}
+
+// Do any third-party scripts want in on the fun?
+if (!defined('SMF_INSTALLING') && class_exists(Config::class, false)) {
+	if (!class_exists(IntegrationHook::class, false) && is_file($sourcedir . '/IntegrationHook.php')) {
+		require_once $sourcedir . '/IntegrationHook.php';
+	}
+
+	if (class_exists(IntegrationHook::class, false)) {
+		IntegrationHook::call('integrate_autoload', [&$third_party_mappers]);
+	}
+}
+
+foreach ($third_party_mappers as $prefix => $dirname) {
+	$loader->addPsr4($prefix, $dirname);
+}
```

Comment:
I'd prefer to see this code moved back into an autoloader class rather than having it here in index.php. 

Part of the reason for moving the existing autoloading code into a separate class was to avoid polluting the global namespace with variables in index.php. (Note: when backward compatibility support we end up putting references to class variables from SMF's namespace into the global namespace anyway, but the idea is to eliminate that in a future version.)